services/project-files-provider-base.ts

```diff
@@ -9,14 +10,15 @@ export abstract class ProjectFilesProviderBase implements IProjectFilesProvider
 		protected $options: ICommonOptions) {}
 
 	public getPreparedFilePath(filePath: string): string {
-		let projectFileInfo = this.getProjectFileInfo(filePath);
+		let projectFileInfo = this.getProjectFileInfo(filePath, "");
 		return path.join(path.dirname(filePath), projectFileInfo.onDeviceFileName);
 	}
 
-	public getProjectFileInfo(filePath: string, platform?: string): IProjectFileInfo {
+	public getProjectFileInfo(filePath: string, platform: string, projectFilesConfig?: IProjectFilesConfig): IProjectFileInfo {
 		let parsed = this.parseFile(filePath, this.$mobileHelper.platformNames, platform || "");
+		let basicConfigurations = [Configurations.Debug.toLowerCase(), Configurations.Release.toLowerCase()];
 		if (!parsed) {
-			parsed = this.parseFile(filePath, ["debug", "release"], this.$options.release ? "release" : "debug");
+			parsed = this.parseFile(filePath, basicConfigurations.concat(projectFilesConfig && projectFilesConfig.additionalConfigurations || []), projectFilesConfig && projectFilesConfig.configuration || basicConfigurations[0]);
```

Comment:
can you extract the `basicConfigurations.concat(projectFilesConfig && projectFilesConfig.additionalConfigurations || []` and `projectFilesConfig && projectFilesConfig.configuration || basicConfigurations[0]` in separate variables in order to simplify the call to `parseFile`